### PR TITLE
Graph navigation and assorted tidying

### LINKED ISSUE #42: Zooming in Graphs
Need a "standard" set of zoom tools for the graphs - principally, a mechanism to draw a box (I believe QtChart calls these rubber bands) and zoom in to that area. Conversely, we'll need a hotkey or similar (e.g. context menu) to allow resetting the view to it's original, "show all" state.

### LINKED ISSUE #49: Make search controls less visible
At present we have a filtering mechanism (above the data table) and a searching mechanism (below the data table) and which need to be differentiated to avoid confusion.

I would suggest the following:
1. Remove the search line edit and the arrow and `searchAll` buttons from the main UI.
2. Add an `Edit` menu with a `Find...` option which uses `QInputDialog::getText` to get the search text and store it in some internal private variable
3. Add `Find Next` and `Find Previous` options to the `Edit` menu (with the accepted shortcuts `F3` and `Shift-F3` respectively) to replicate the navigation arrow buttons.
4. Give feedback on the search (number of items, "Search hit bottom..." etc.) in a label on the window's status bar.

### LINKED ISSUE #61: Absolute time is relative to epoch?
The absolute time plotting currently gives us basically the same view as the time/date plotting. What should really happen is that data should be plotted on an axis (in seconds) with timings relative to the individual run start times, making all plots overlap to some degree.

### LINKED ISSUE #62: Tidy graph widget
Additional controls for the graph widget are currently at the lower corners of the plot and overlap the main graph.  These should be moved above the `ChartView` and placed in a "toolbar" of some sort.

### PR DESCRIPTION
**Pull request by EthanDevlinSTFC**
Created container widget for search elements to tidy code
Minor bug fix for search label
Creating custom chartview class to enable more detailed and personalised navigation closes #42 

(Discussion over specific graph features and navigation style to be had)

closes #62 
closes #61

### CODE REVIEW

**Comment by EthanDevlinSTFC** [comment]
Ctrl + F = find, focus on find if open
Ctrl + Shift + F = close find
F3, Shift + F3 = cycle search results

graphing:
Ctrl = hold to toggle zoom dimensions (for rubber band)
arrow keys to move
mouse wheel to zoom
mouse wheel click to reset

**Review by trisyoungs — CHANGES_REQUESTED** [review]
Search navigation for the run data works really nicely.  Rather than a set of (hidden) controls at the foot of the window, however, I'd rather see this implemented in a more standard way as a modal dialog raised with `Ctrl-F`.  Once a `Find` button is clicked (or a `Select All` button to replicate the functionality of your `searchAll` button) the dialog can close and we then rely just on `F3` and `Shift-F3` for navigation. Displaying the number of search results can then be a job for a permanent widget on the statusbar instead (which at present is sitting there taking up space and not doing anything!

Couple of other observations:
1. I don't seem to have a rubberband zoom at all
2. I like the idea of a middle-mouse click resetting the graph, but it turns out my fingers are used to clicking and holding the middle mouse button to interactively translate the view. This is a feature I implemented in my own graphing stuff, so it isn't what you might call standard, but you could consider going that route as well.

Some other thoughts occurred while I was playing around with the code, which I'll capture as new issues.

**Comment by EthanDevlinSTFC** [comment]
Strange issue where the rubber band suddenly stopped working. I've reimplemented the way it works so hopefully there won't be another issue there. 
Graph translation on middle click is in, I've prevented the default zoom out on right click because it seems quite bad to me and have used that to replace zoom reset.

File: frontend/mainwindow.cpp [hunk 1]
```diff
@@ -133,48 +133,40 @@ void MainWindow::on_massSearchButton_clicked()
...
     }
+    if (event->key() == Qt::Key_F3 && event->modifiers() == Qt::ShiftModifier)
+    {
+        on_findUp_clicked();
+        return;
+    }
+    if (event->key() == Qt::Key_F3)
+        on_findDown_clicked();
+
+    // SWITCH STATEMENT
```
Comment (trisyoungs) [inline frontend/mainwindow.cpp]: I assume this is a "note to self" to convert this to a switch statement. Given that you're checking combinations of keys and modifiers, a set of `if`/`else if`s is probably neater here.

File: frontend/chartview.cpp [hunk 2]
```diff
@@ -0,0 +1,113 @@
...
+void ChartView::mousePressEvent(QMouseEvent *event)
+{
+    if (event->button() == Qt::MiddleButton)
+    {
+        if (first_)
+        {
+            chart()->zoomIn();
+            chart()->zoomOut();
+            first_ = false;
+        }
```
Comment (trisyoungs) [inline frontend/chartview.cpp]: I'm sure this is a Qt-related intricacy, but why the need to treat the first middle-click as a special case?

File: frontend/nexusInteraction.cpp [hunk 3]
```diff
@@ -289,15 +315,15 @@ void MainWindow::toggleAxis(int state)
...
-        help[0]->chart()->axes()[1]->show();
+        tabCharts[0]->hide();
+        tabCharts[1]->show();
     }
     else
     {
-        help[0]->chart()->axes()[0]->show();
-        help[0]->chart()->axes()[1]->hide();
+        tabCharts[0]->show();
+        tabCharts[1]->hide();
```
Comment (trisyoungs) [inline frontend/nexusInteraction.cpp:327]: Controlling the widgets in this way is valid, but a bit dangerous and hacky and another good reason to have a compound widget and reusable UI file representing all the controls.

**Review by trisyoungs — COMMENTED** [review]
I think I quite like the separation of the rubberband zoom into x and (Ctrl-) y actions - however, it remains to be seen whether the users will accept it over the more traditional rectangle zoom!

Just a note - this PR doesn't really close #49 since that issue requires taking the search controls off the main UI entirely and putting them in a dialog.

Like where all this is going - some more small things I'll capture as further issues.

File: frontend/mainwindow.cpp (hunk 1 above)
Comment (EthanDevlinSTFC) [reply frontend/mainwindow.cpp]: Noticed this, just forgot to remove the comment

File: frontend/chartview.cpp (hunk 2 above)
Comment (EthanDevlinSTFC) [reply frontend/chartview.cpp]: for the reset to reset scrolling a zoom event needs to occur beforehand. I can remove the zoom from reset and just have it always do on middle click which is cleaner

File: frontend/nexusInteraction.cpp (hunk 3 above)
Comment (EthanDevlinSTFC) [reply frontend/nexusInteraction.cpp:327]: Definitely need to get it converted to a compound widget. I think this is best captured as a separate issue.

**Review by trisyoungs — APPROVED** [review]

**Pull request merged**